pydantic/analyzed_type.py

```diff
@@ -0,0 +1,241 @@
+from __future__ import annotations
+
+import sys
+from typing import TYPE_CHECKING, Any, Dict, Generic, Iterable, Set, TypeVar, Union, overload
+
+from pydantic_core import CoreConfig, CoreSchema, SchemaSerializer, SchemaValidator, core_schema
+from typing_extensions import Literal
+
+from pydantic.config import ConfigDict
+from pydantic.json_schema import DEFAULT_REF_TEMPLATE, GenerateJsonSchema
+
+from ._internal import (
+    _generate_schema,
+    _typing_extra,
+)
+
+T = TypeVar('T')
+
+if TYPE_CHECKING:
+    # should be `set[int] | set[str] | dict[int, IncEx] | dict[str, IncEx] | None`, but mypy can't cope
+    IncEx = Union[Set[int], Set[str], Dict[int, Any], Dict[str, Any]]
+
+
+def _get_schema(type_: Any, config: CoreConfig | None, parent_depth: int) -> CoreSchema:
+    arbitrary_types = bool((config or {}).get('arbitrary_types_allowed', False))
+    local_ns = _typing_extra.parent_frame_namespace(parent_depth=parent_depth)
+    # BaseModel uses it's own __module__ to find out where it was defined
```

Comment:
very good, might be easier to read as a docstring, even if we don't publish it in documentation.